src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java

```diff
@@ -985,226 +979,99 @@ private String getPermissionFilterQuery(DataverseRequest dataverseRequest, SolrQ
-
-//            /**
-//             * @todo add onlyDatatRelatedToMe option into the experimental JOIN
-//             * before enabling it.
-//             */
-        /**
-         * From a search perspective, we don't care about if the group was
-         * created within one dataverse or another. We just want a list of *all*
-         * the groups the user is part of. We are greedy. We want all BuiltIn
-         * Groups, Shibboleth Groups, IP Groups, "system" groups, everything.
-         *
-         * A JOIN on "permission documents" will determine if the user can find
-         * a given "content document" (dataset version, etc) in Solr.
-         */
-        String groupsFromProviders = "";
-        StringBuilder sb = new StringBuilder();
         for (Group group : groups) {
-            logger.fine("found group " + group.getIdentifier() + " with alias " + group.getAlias());
             String groupAlias = group.getAlias();
-            if (groupAlias != null && !groupAlias.isEmpty()) {
-                sb.append(" OR ");
-                // i.e. group_builtIn/all-users, group_builtIn/authenticated-users, group_1-explictGroup1, group_shib/2
-                sb.append(IndexServiceBean.getGroupPrefix() + groupAlias);
+            if (groupAlias != null && !groupAlias.isEmpty() && (!avoidJoin || !groupAlias.startsWith("builtIn"))) {
+                groupList.add(IndexServiceBean.getGroupPrefix() + groupAlias);
             }
         }
-        groupsFromProviders = sb.toString();
 
-        logger.fine(groupsFromProviders);
-        if (true) {
-            /**
-             * @todo get rid of "experimental" in name
-             */
-            String experimentalJoin = "{!join from=" + SearchFields.DEFINITION_POINT + " to=id}" + SearchFields.DISCOVERABLE_BY + ":(" + IndexServiceBean.getPublicGroupString() + " OR " + IndexServiceBean.getGroupPerUserPrefix() + au.getId() + groupsFromProviders + ")";
-            publicPlusUserPrivateGroup = experimentalJoin;
+        if (!avoidJoin) {
+            // Add the public group
+            groupList.add(0, IndexServiceBean.getPublicGroupString());
+        } 
+        
+        String groupString = null;
+        //If we have additional groups, format them correctly into a search string, with parens if there is more than one
+        if (groupList.size() > 1) {
+            groupString = "(" + StringUtils.join(groupList, " OR ") + ")";
+        } else if (groupList.size() == 1) {
+            groupString = groupList.get(0);
         }
-
-        //permissionFilterQuery = publicPlusUserPrivateGroup;
-        logger.fine(publicPlusUserPrivateGroup);
-
-        return publicPlusUserPrivateGroup;
-
+        logger.fine("Groups: " + groupString);
+        String permissionQuery = buildPermissionFilterQuery(avoidJoin, groupString);
+        logger.fine("Permission Query: " + permissionQuery);
```

Comment:
This and the query from the existing code line 1203 should be the same in all cases, whether the avoidJoin flag is set or not.